mods/hv/rules/buildings.yaml

```
@@ -1852,6 +1852,106 @@ BARRIER:
 	-Selectable:
 	Interactable:
 
+^GATE:
+	Inherits: ^Building
+	Inherits@Handicaps: ^PlayerHandicaps
+	Building:
+		RequiresBaseProvider: true
+		TerrainTypes: Clear, Road, Crater, Grass, Grass Pit, Snow, Red Snow, Sand, Stone, Tech
+		AllowInvalidPlacement: false
+		BuildSounds: Quedicemipez_teleportation-barrier.wav
+	Buildable:
+		BuildPaletteOrder: 15
+		Queue: Defense
+		Prerequisites: module
+		Description: Automated barrier that opens for allied units.
+	Valued:
+		Cost: 300
+	Health:
+		HP: 75000
+	Armor:
+		Type: Concrete
+	LineBuildNode:
+	MustBeDestroyed:
+		RequiredForShortGame: false
+	-WithSpriteBody:
+	WithGateSpriteBody:
+		OpenSequence: open
+	Tooltip:
+		Name: Gate
+	RequiresBuildableArea:
+		AreaTypes: building
+		Adjacent: 6
+	Targetable:
+		TargetTypes: Ground, Wall, NoAutoTarget
+	RenderSprites:
+		PlayerPalette: green
+	MapEditorData:
+		Categories: Wall
+	Gate:
+		OpeningSound: lowpolygon_gate-in.wav
+		ClosingSound: lowpolygon_gate-out.wav
+	BlocksProjectiles:
+	OwnerLostAction:
+		Action: ChangeOwner
+	ThrowsShrapnel:
+		Weapons: Debris
+		Pieces: 2, 3
+		Range: 1c512, 3c0
+	ThrowsShrapnel@large:
+		Weapons: LargeDebris
+		Pieces: 0, 1
+		Range: 1c512, 3c0
+	Demolishable:
+	Explodes:
+		Weapon: UnitExplodeLarge
+		EmptyWeapon: UnitExplodeLarge
+	MapEditorData:
+		Categories: Wall
+	-ShakeOnDeath:
+	WithConstructionBeamOverlay:
+	UpdatesPlayerStatistics:
+		AddToAssetsValue: false
+	WithShadow:
+		Offset: 200,200,0
+		ZOffset: -1000
+	Sellable:
+		SellSounds: Quedicemipez_teleportation-barrier-sell.wav
+
+GATE-A:
```

I would prefer something a bit more descriptive like
```suggestion
GATE-EW:
```

---

or maybe `VGATE/HGATE`